src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1069,919 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+            for f in older_version_files:
+                try:
+                    os.remove(f)
+                except:
+                    logger.warning("failed to delete older version files")
+
+        try:
+            with open(install_location, 'wb') as f:
+                f.write(responseContent)
+        except Exception as e:
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception=e, fault_type=consts.Create_CSPExe_Fault_Type,
+                                    summary='Unable to create proxy executable')
+            raise CLIError("Failed to create proxy executable." + str(e))
+
+        os.chmod(install_location, os.stat(install_location).st_mode | stat.S_IXUSR)
+
+    # Creating config file to pass config to clientproxy
+    config_file_location = os.path.join(install_dir, 'config.yml')
+
+    if os.path.isfile(config_file_location):
+        try:
+            os.remove(config_file_location)
+        except Exception as e:
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception=e, fault_type=consts.Remove_Config_Fault_Type,
+                                    summary='Unable to remove old config file')
+            raise CLIError("Failed to remove old config." + str(e))
+
+    # initializations
+    user_type = 'sat'
+    creds = ''
+
+    # if service account token is not passed
+    if token is None:
+        # Identifying type of logged in entity
+        account = get_subscription_id(cmd.cli_ctx)
+        account = Profile().get_subscription(account)
+        user_type = account['user']['type']
+
+        tenantId = _graph_client_factory(cmd.cli_ctx).config.tenant_id
+
+        if user_type == 'user':
+            dict_file = {'server': {'httpPort': int(client_proxy_port), 'httpsPort': int(api_server_port)}, 'identity': {'tenantID': tenantId, 'clientID': consts.CLIENTPROXY_CLIENT_ID}}
+        else:
+            dict_file = {'server': {'httpPort': int(client_proxy_port), 'httpsPort': int(api_server_port)}, 'identity': {'tenantID': tenantId, 'clientID': account['user']['name']}}
+
+        if cloud == 'DOGFOOD':
+            dict_file['cloud'] = 'AzureDogFood'
+
+        # Fetching creds
+        creds_location = os.path.expanduser(os.path.join('~', creds_string))
+        try:
+            with open(creds_location) as f:
+                creds_list = json.load(f)
+        except Exception as e:
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception=e, fault_type=consts.Load_Creds_Fault_Type,
+                                    summary='Unable to load accessToken.json')
+            raise CLIError("Failed to load credentials." + str(e))
+
+        user_name = account['user']['name']
+        loop_flag = False
+        for i in range(len(creds_list)):
+            if loop_flag:
+                break
+            creds_obj = creds_list[i]
+            for key in creds_obj:
+                if user_type == 'user':
```

Comment:
ideally ideally if creds_obj[key] == user_name: should be checked immediately when we enter into the for loop

Reply:
thanks for pointing out. I have refactored this part.